src/callbacks.jl

```diff
@@ -0,0 +1,70 @@
+"""
+    Callbacks(; lazy = nothing, heuristic = nothing)
+
+Set MIP-based callbacks. Two basic types are offered: `lazy` and `heuristic`.
+
+### The lazy callback
+
+The `lazy` callback can be used to add *lazy* constraints which are added on
+demand at integer nodes in the branch and bound tree.
+
+- The lazy constraint callback *may* be called when the solver has an integer
+  primal solution
+- You can only access the primal solution through `VariablePrimal`. That is,
+  `ConstraintDual` etc will not work
+- When the solver terminates, it will provide a solution that satisfies all
+  lazy constraints
+- We make no claims about what happens to solutions during the B&B tree
+
+See also [`add_lazy_constraint`](@ref).
+
+### The heuristic callback
+
+The `heuristic` callback can be used to provide the solver with heuristically
+obtained integer-feasible solutions at fractional nodes in the branch and bound
+tree.
+
+- The heuristic callback *may* be called when the solver has a fractional (i.e.,
+  non-integer) solution
+- The solver may silently reject the provided solution
+- Some solvers require a complete solution, others only partial solutions. It's
```

Comment:
Does the solution have to satisfy all constraints in the model?

Reply:
No, see above

Reply:
If a solution does not satisfy constraints, is it guaranteed to be rejected?